packages/mobile/src/screens/favorites-screen/TracksTab.tsx

```diff
@@ -146,47 +224,60 @@ export const TracksTab = () => {
     [playingUid, isPlaying, dispatch, isReachable, isOfflineModeEnabled]
   )
 
-  const isLoading = savedTracksStatus !== Status.SUCCESS
-  const tracks = savedTracks.entries as TrackMetadata[]
-  const hasNoFavorites = tracks.length === 0
+  const filteredTracks = allTracksFetched
+    ? tracks.filter(filterTrack)
+    : tracks.filter((track) => track.kind !== Kind.EMPTY)
 
   return (
-    <WithLoader loading={isLoading}>
-      <VirtualizedScrollView listKey='favorites-screen'>
-        {!isLoading && hasNoFavorites && !filterValue ? (
-          isOfflineModeEnabled && !isReachable ? (
-            <NoTracksPlaceholder />
-          ) : (
-            <EmptyTileCTA message={messages.emptyTabText} />
-          )
+    <VirtualizedScrollView listKey='favorites-screen'>
+      {!isLoading && tracks.length === 0 && !filterValue ? (
+        isOfflineModeEnabled && !isReachable ? (
+          <NoTracksPlaceholder />
         ) : (
-          <>
-            <OfflineContentBanner />
-            <FilterInput
-              value={filterValue}
-              placeholder={messages.inputPlaceholder}
-              onChangeText={setFilterValue}
-            />
-            {tracks.length ? (
-              <Tile
-                styles={{
-                  root: styles.container,
-                  tile: styles.trackListContainer
-                }}
-              >
-                <TrackList
-                  onSave={onToggleSave}
-                  showDivider
-                  togglePlay={togglePlay}
-                  trackItemAction='save'
-                  tracks={tracks.filter(filterTrack)}
-                  hideArt
+          <EmptyTileCTA message={messages.emptyTabText} />
+        )
+      ) : (
+        <>
+          <OfflineContentBanner />
```

Comment:
Can all this be part of the flat list header? Also I wonder if we can remove the virtualized scroll view then